com.unity.render-pipelines.universal/Runtime/Passes/DrawObjectsPass.cs

```diff
@@ -54,6 +57,29 @@ internal DrawObjectsPass(URPProfileId profileId, bool opaque, RenderPassEvent ev
             m_ProfilingSampler = ProfilingSampler.Get(profileId);
         }
 
+        public void SetUseDepthPriming(bool useDepthPriming)
+        {
+            m_UseDepthPriming = useDepthPriming;
+        }
+
+        public override void Configure(CommandBuffer cmd, RenderTextureDescriptor cameraTextureDescriptor)
+        {
+            if (m_UseDepthPriming)
+            {
+                m_RenderStateBlock.depthState = new DepthState(false, CompareFunction.Equal);
+                m_RenderStateBlock.mask |= RenderStateMask.Depth;
+                m_PreviouslyUsedDepthPriming = true;
+            }
+            // If the previous execution of the pass was using depth priming, only then reset the depth state to default.
+            // Always resetting to default would break the custom state set by certain DrawObjectPass instances such as particle rendering.
+            else if (m_PreviouslyUsedDepthPriming)
```

Comment:
Hmm, it feels like this is hiding an underlying issue. What's the actual problem going on here? I'd like to avoid this statefulness.

Reply:
Luckily it looks like this is no longer relevant. Adding a DepthOnly pass to the custom shader of the particles test scene seems to resolve this. Wasn't very fond of this state management at all so glad it's gone.

Reply:
Ahh okay, great to hear! I did have a hard time figuring out when it actually hit the second state :p